web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/BaseCreateDraftCertificateValidator.java

```diff
@@ -83,13 +83,32 @@ protected void validatePUServiceResponse(ResultValidator errors, Personnummer pe
 
     protected void validateBusinessRulesForSekretessmarkeradPatient(ResultValidator errors, Personnummer personnummer, String intygsTyp,
         IntygUser user) {
-        if (personnummer != null) {
-            final SekretessStatus sekretessStatus = patientDetailsResolver.getSekretessStatus(personnummer);
-            if (sekretessStatus != SekretessStatus.UNDEFINED) {
-                validateSekretess(errors, intygsTyp, sekretessStatus);
-                validateHsaUserMayCreateDraft(errors, user, sekretessStatus);
-            }
+        if (personnummer == null) {
+            return;
+        }
+
+        final var sekretessStatus = patientDetailsResolver.getSekretessStatus(personnummer);
+        if (sekretessStatus == SekretessStatus.FALSE) {
+            return;
         }
+
+        final var certificateAllowedForSekretess = authoritiesHelper.getIntygstyperAllowedForSekretessmarkering().contains(intygsTyp);
```

Comment:
Do we have a logical error here?

If we can't retrieve the patient from PU, then it will only give an error if it is not possible to create the certificate type for "sekretess"-patient.

It is good that you check above if it is SekretessStatus.FALSE and just do return.

If it is SekretessStatus.UNKNOWN (typing from memory, so I might have the enum-name wrong), then you should give the error that is produced now on line 110.

If it is SekretessStatus.TRUE, then it should first check if it is possible to create that type on "sekretess"-patient. If not, give the error on line 107. If it is OK, then continue to validateHsaUserMayCreateDraft.